test/s3select_test.cpp

```diff
@@ -238,3 +238,33 @@ TEST(TestS3SElect, utcnow_function)
     ASSERT_EQ(s3select_res, boost::posix_time::to_simple_string(now));
 }
 
+void generate_csv(std::string& out, size_t size) {
+  // schema is: int, float, string, string
+  std::stringstream ss;
+  for (auto i = 0U; i < size; ++i) {
+    ss << i << "," << i/10.0 << "," << "foo"+std::to_string(i) << "," << std::to_string(i)+"bar" << std::endl;
+  }
+  out = ss.str();
+}
+
+TEST(TestS3selectFunctions, charlength)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select charlength(\"abcde\") from stdin;";
```

Comment:
as noted (other PR comment) , we should extend those kind of test.

please add empty string test.

also add /characterlength/ test call, to verified s3select syntax parser support different names for identical operation.

Reply:
Updated the pr. Please take a look